test/distro-qemu/build-and-run.sh

```diff
@@ -85,11 +95,16 @@ if [[ "${USE_DOCKER}" == "1" ]] && command -v docker &> /dev/null; then
                  wget -q https://go.dev/dl/go1.24.7.linux-${GOARCH}.tar.gz && \
                  tar -C /usr/local -xzf go1.24.7.linux-${GOARCH}.tar.gz && \
                  export PATH=/usr/local/go/bin:\$PATH && \
-                 CGO_ENABLED=1 go test -c ../../interpreter/rtld ../../support/usdt"
+                 CGO_ENABLED=1 go test -c ../../interpreter/rtld ../../support/usdt/test"
 else
     # Local build with cross-compilation if needed
     echo "Building locally for ${GOARCH}..."
-    CGO_ENABLED=1 GOARCH=${GOARCH} go test -c ../../interpreter/rtld ../../support/usdt
+    if [ "$GOARCH" = "arm64" ]; then
+        # Cross-compile for ARM64 using aarch64-linux-gnu-gcc
+        CGO_ENABLED=1 GOARCH=${GOARCH} CC=aarch64-linux-gnu-gcc go test -c ../../interpreter/rtld ../../support/usdt/test
+    else
+        CGO_ENABLED=1 GOARCH=${GOARCH} go test -c ../../interpreter/rtld ../../support/usdt/test
+    fi
```

Comment:
Can you check if the arch is x86 here and fail otherwise? I want to limit introducing new cases where we assume the only arches in the world are arm and x86 (because that will make it easier to support riscv in the future)